data-formats/src/messages/diun.rs

```diff
@@ -0,0 +1,253 @@
+use serde::{Deserialize, Serialize};
+use serde_tuple::Serialize_tuple;
+
+use super::{ClientMessage, EncryptionRequirement, Message, ServerMessage};
+use crate::{
+    constants::{KeyStorageType, MessageType, MfgStringType, PublicKeyType},
+    publickey::PublicKey,
+    types::{COSESign, CipherSuite, KexSuite, Nonce},
+};
+
+#[derive(Debug, Serialize_tuple, Deserialize)]
+pub struct Connect {
+    nonce_diun_1: Nonce,
+    kex_suite: KexSuite,
+    cipher_suite: CipherSuite,
+    key_exchange: Vec<u8>,
+}
+
+impl Connect {
+    pub fn new(
+        nonce_diun_1: Nonce,
+        kex_suite: KexSuite,
+        cipher_suite: CipherSuite,
+        key_exchange: Vec<u8>,
+    ) -> Self {
+        Connect {
+            nonce_diun_1,
+            kex_suite,
+            cipher_suite,
+            key_exchange,
+        }
+    }
+
+    pub fn nonce_diun_1(&self) -> &Nonce {
+        &self.nonce_diun_1
+    }
+
+    pub fn kex_suite(&self) -> &KexSuite {
+        &self.kex_suite
+    }
+
+    pub fn cipher_suite(&self) -> &CipherSuite {
+        &self.cipher_suite
+    }
+
+    pub fn key_exchange(&self) -> &[u8] {
+        &self.key_exchange
+    }
+}
+
+impl Message for Connect {
+    fn message_type() -> MessageType {
+        MessageType::DIUNConnect
+    }
+
+    fn is_valid_previous_message(message_type: Option<MessageType>) -> bool {
+        matches!(message_type, None)
```

Comment:
I understand that what we're ensuring here is that there's no previous message?

Reply:
Yep. This is the start of an interaction, so no messages should have been before this during this session.